### PR TITLE
feat: support wiremock fault injection

### PR DESCRIPTION
**Pull request by omerlh**
https://wiremock.org/docs/simulating-faults/

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -19,14 +19,20 @@ export interface RequestPattern {
...
+interface SuccessResponse {
   status: number;
   jsonBody?: Record<string, any>;
   headers?: Record<string, any>;
   body?: string;
   base64Body?: string;
 }
 
+interface FailureResponse {
+  fault: 'CONNECTION_RESET_BY_PEER' | 'EMPTY_RESPONSE' | 'MALFORMED_RESPONSE_CHUNK' | 'RANDOM_DATA_THEN_CLOSE';
```
Comment (SharonGrossman) [inline src/types.ts:31]: are those faults documented anywhere?

Comment (SharonGrossman) [reply src/types.ts:31]: I mean, if they are standardized

Comment (omerlh) [reply src/types.ts:31]: See the link on the PR :)

**Review by danielwitz — COMMENTED** [review]
Can you add this use case to the read.me?

**Comment by omerlh** [comment]
Sure, added

**Review by danielwitz — APPROVED** [review]

**Pull request merged**